ntcore/src/main/native/cpp/server/ServerClientLocal.cpp

```diff
@@ -8,6 +8,27 @@
 
 using namespace nt::server;
 
+// False positive from GCC 14:
+//
+// clang-format off
+//   In copy constructor ‘std::function<_Res(_ArgTypes ...)>::function(const std::function<_Res(_ArgTypes ...)>&) [with _Res = void; _ArgTypes = {unsigned int}]’,
+//       inlined from ‘nt::server::ServerClient4Base::ServerClient4Base(std::string_view, std::string_view, bool, nt::server::SetPeriodicFunc, nt::server::ServerStorage&, int, wpi::Logger&)’ at /home/tav/frc/wpilib/allwpilib/ntcore/src/main/native/cpp/server/ServerClient4Base.h:24:77,
+//       inlined from ‘nt::server::ServerClientLocal::ServerClientLocal(nt::server::ServerStorage&, int, wpi::Logger&)’ at /home/tav/frc/wpilib/allwpilib/ntcore/src/main/native/cpp/server/ServerClientLocal.cpp:18:75:
+//   /usr/include/c++/14.2.1/bits/std_function.h:391:17: error: ‘<anonymous>’ may be used uninitialized [-Werror=maybe-uninitialized]
+//     391 |             __x._M_manager(_M_functor, __x._M_functor, __clone_functor);
+//         |             ~~~~^~~~~~~~~~
+//   /usr/include/c++/14.2.1/bits/std_function.h: In constructor ‘nt::server::ServerClientLocal::ServerClientLocal(nt::server::ServerStorage&, int, wpi::Logger&)’:
+//   /usr/include/c++/14.2.1/bits/std_function.h:267:7: note: by argument 2 of type ‘const std::_Any_data&’ to ‘static bool std::_Function_handler<_Res(_ArgTypes ...), _Functor>::_M_manager(std::_Any_data&, const std::_Any_data&, std::_Manager_operation) [with _Res = void; _Functor = nt::server::ServerClientLocal::ServerClientLocal(nt::server::ServerStorage&, int, wpi::Logger&)::<lambda(uint32_t)>; _ArgTypes = {unsigned int}]’ declared here
+//     267 |       _M_manager(_Any_data& __dest, const _Any_data& __source,
+//         |       ^~~~~~~~~~
+//   /home/tav/frc/wpilib/allwpilib/ntcore/src/main/native/cpp/server/ServerClientLocal.cpp:18:75: note: ‘<anonymous>’ declared here
+//      18 |     : ServerClient4Base{"", "", true, [](uint32_t) {}, storage, id, logger} {
+//         |                                                                           ^
+// clang-format on
+#if defined(__GNUC__) && !defined(__clang__)
```

Comment:
Let’s shorten this comment to simply say it suppresses a false positive.  We can put the details in the commit message.